### PR TITLE
Table column filter could be ReactNode

### PR DESCRIPTION
**Pull request by vinhnph-ibl**
First of all, thank you for your contribution! 😄

New feature please send pull request to feature branch, and rest to master branch.
Pull request will be merged after one of collaborators approve.
Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Component style update
- [X] TypeScript definition update
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?

> 1. Describe the source of requirement.
While using typescript, `ColumnFilterItem` text type of Table is valid only by string. It should be either string or ReactNode 
> 2. Resolve what problem.
Filter item could be string or ReactNode
> 3. Related issue link.

### API Realization (Optional if not new feature)

> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.

### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Changelog description (Optional if not new feature)

> 1. English description
> 2. Chinese description (optional)

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [X] TypeScript definition is updated/provided or not needed
- [ ] Changelog is provided or not needed

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

**Comment by vinhnph-ibl** [comment]
Before and after

**Comment by vinhnph-ibl** [comment]
Please have a review here, @afc163

File: components/table/interface.tsx [hunk 1]
```diff
@@ -6,18 +6,17 @@ import { CheckboxChangeEvent } from '../checkbox';
 import { PaginationConfig } from '../pagination';
 export { PaginationConfig } from '../pagination';
 
-export type CompareFn<T> = ((a: T, b: T, sortOrder?: SortOrder) => number);
-export type ColumnFilterItem = { text: string; value: string; children?: ColumnFilterItem[] };
+export type CompareFn<T> = (a: T, b: T, sortOrder?: SortOrder) => number;
+export type ColumnFilterItem = {
+  text: string | React.ReactNode;
```
Comment (afc163) [inline components/table/interface.tsx]: `React.ReactNode` contains `string`.

Comment (yesmeck) [inline components/table/interface.tsx]: Just `text: React.ReactNode`, `React.ReactNode` has already included `string`.

File: components/table/interface.tsx [hunk 2]
```diff
@@ -6,18 +6,17 @@ import { CheckboxChangeEvent } from '../checkbox';
...
 export interface ColumnProps<T> {
   title?:
     | React.ReactNode
-    | ((
-        options: {
-          filters: TableStateFilters;
-          sortOrder?: SortOrder;
-        },
-      ) => React.ReactNode);
+    | ((options: { filters: TableStateFilters; sortOrder?: SortOrder }) => React.ReactNode);
```
Comment (yesmeck) [inline components/table/interface.tsx]: Why these style changes?

Comment (vinhnph-ibl) [reply components/table/interface.tsx]: It's eslint automatically fixed

**Review by yesmeck — APPROVED** [review]

**Pull request merged**